File: src/spacerocks/calc_E_from_f.cpp
#include "spacerocks.hpp"

double calc_E_from_f(double e, double f) {

  double E;

  if (e < 1) {
    E = 2 * atan2(sqrt(1 - e) * sin(f / 2), sqrt(1 + e) * cos(f / 2));
  }
  else {
    double cta = cos(f);
    E = acosh((cta + e) / (1 + e * cta));
    if (f < 0) {
      E *= -1;
    }
  }
  return E;
}

File: src/transforms/calc_E_from_f.rs

#![allow(non_snake_case)]
pub fn calc_E_from_f(e: f64, f: f64) -> f64 {
    let mut E;

    if e < 1.0 {
        E = 2.0 * ((1.0 - e).sqrt() * (f / 2.0).sin()).atan2((1.0 + e).sqrt() * (f / 2.0).cos())
    }

    else {
        // let cta = f.cos();
        // E = ((cta + e) / (1.0 + e * cta)).acosh();
        // if f < 0.0 {
        //     E *= -1.0;
        // }

        E = 2.0 * (((e - 1.0) / (e + 1.0)).sqrt() * (f / 2.0).tan()).atanh();
        


    }
    return E;
}


#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_calc_E_from_f() {
        let e = 0.0;
        let f = 0.0;
        let result = calc_E_from_f(e, f);
        assert_eq!(result, 0.0);
    }
}
